mdb-realm/src/helpers.ts

```diff
@@ -178,91 +90,93 @@ export async function waitForClusterDeployment(config: EnvironmentConfig): Promi
     throw new Error(`Cluster failed to deploy after ${100 * pollDelay} seconds`);
 }
 
-export async function configureRealmCli(config: EnvironmentConfig): Promise<void> {
-    await execCmd("npm i -g mongodb-realm-cli");
-
-    await execCliCmd(
-        `login --api-key ${config.apiKey} --private-api-key ${config.privateApiKey} --atlas-url ${config.atlasUrl} --realm-url ${config.realmUrl}`,
-    );
-}
-
-export async function publishApplication(appPath: string, config: EnvironmentConfig): Promise<{ id: string }> {
-    const appName = `${path.basename(appPath)}-${getSuffix()}`;
-    core.info(`Creating app ${appName}`);
-
-    const createResponse = await execCliCmd(`apps create --name ${appName}`);
-
-    const appId = createResponse.map(r => r.doc).find(d => d && d.client_app_id).client_app_id;
-
-    core.info(`Created app ${appName} with Id: ${appId}`);
+export async function deleteApps(config: EnvironmentConfig, deleteAll = false): Promise<void> {
+    const accessToken = await authenticate(config);
 
-    const secrets = readJson(path.join(appPath, "secrets.json"));
+    const listResponse = await execRealmRequest("GET", "apps", accessToken, config);
+    const allApps = (listResponse as any[])
+        .map(a => {
+            return { name: a.name, id: a._id };
+        })
+        .filter(a => deleteAll || a.name.includes(config.clusterName));
```

Comment:
@nirinchev Why not using the check by app.dataSource that you have introduced in [Delete apps that don't have datasource defined](https://github.com/realm/ci-actions/pull/6/files#top)?
```
const dataSource = describeResponse[0]?.doc.data_sources[0]?.data_source;
if (!dataSource || dataSource === config.clusterName) {
```

Reply:
Because in the doc we agreed to use the app naming convention, so if an app includes the cluster name, we delete it regardless of whether it links to the cluster or not.

Reply:
OK